etcd/v3/kv_etcd.go

```diff
@@ -700,9 +700,16 @@ func (et *etcdKV) Unlock(kvp *kvdb.KVPair) error {
 	l.Lock()
 	// Don't modify kvp here, CompareAndDelete does that.
 	_, err := et.CompareAndDelete(kvp, kvdb.KVFlags(0))
-	if err == nil {
+	connectionError := false
```

Comment:
If etcd is genuinely down, CompareAndDelete is going to retry its operation 30 times. This means that the refresh lock will be failing as well. Eventually the lock would expire and CompareAndDelete will get ErrValueMismatch.

Usually that is what we have seen. So why not do this unconditionally? Basically stop the refreshLock routine when an Unlock() is invoked.

Reply:
we don't check refresh status on unlock. if kvdb went down when we are unlocking, we can stop the refresh routine since CompareAndDelete is going to return a connection error.
If kvdb went down and came back and someone acquired the lock we return the error without hiding it (for now).

Reply:
Had an offline discussion with @sangleganesh .